async-nats/src/lib.rs

```diff
@@ -684,7 +695,7 @@ impl Client {
             .await
             .unwrap();
 
-        Ok(Subscriber::new(sid, receiver))
+        Ok(Subscriber::new(sid, self.sender.clone(), receiver))
```

Comment:
We break and shutdown the connector processing task when all the client senders are closed, so this will keep everything alive as long as the subscriber is alive.

This is probably desirable, but note to us we need to document this well.